File: api-py/app/db/models/second_level_task.py
"""二级任务实体(表 second_level_task,camelCase 列)。"""
from datetime import datetime
from typing import Optional

from sqlalchemy import BigInteger, Integer, SmallInteger, String, Text, UniqueConstraint
from sqlalchemy.dialects.mysql import DATETIME
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class SecondLevelTask(Base):
    __tablename__ = "second_level_task"
    # 与真源 project/scripts/schema.sql 的 UNIQUE KEY IDX_827dce5e14d27c57b8bc5d1cb5(taskId) 对齐(仅补 ORM 声明,不改库结构)
    __table_args__ = (UniqueConstraint("taskId", name="IDX_827dce5e14d27c57b8bc5d1cb5"),)

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    taskId: Mapped[str] = mapped_column(String(32))
    firstLevelTaskId: Mapped[str] = mapped_column(String(32))
    stageId: Mapped[str] = mapped_column(String(32))
    title: Mapped[str] = mapped_column(String(128))
    description: Mapped[str] = mapped_column(Text, nullable=True)
    detail: Mapped[str] = mapped_column(Text, nullable=True)
    type: Mapped[str] = mapped_column(String(16))
    completionType: Mapped[str] = mapped_column(String(16))
    actionText: Mapped[Optional[str]] = mapped_column(String(64), nullable=True)
    actionUrl: Mapped[Optional[str]] = mapped_column(String(512), nullable=True)
    # 行为语义字段(schema v1.11 / #DB-23;投影与校验见 #PB-39;真源 §1.3 / §5.2 API-09):
    #   actionType  = 按钮点击后的交互类型(10 值枚举,默认 none)
    #   actionParam = 行为参数(**不透明标识**,仅 data_form/data_upload 使用;其余必须为 NULL)
    actionType: Mapped[str] = mapped_column(String(32), default="none")
    actionParam: Mapped[Optional[str]] = mapped_column(String(64), nullable=True)
    tag: Mapped[Optional[str]] = mapped_column(String(32), nullable=True)
    defaultCompleted: Mapped[int] = mapped_column(SmallInteger, default=0)
    status: Mapped[int] = mapped_column(SmallInteger, default=1)
    sortOrder: Mapped[int] = mapped_column(Integer, default=0)
    createdAt: Mapped[datetime] = mapped_column(DATETIME(fsp=6), default=datetime.now)
    updatedAt: Mapped[datetime] = mapped_column(DATETIME(fsp=6), default=datetime.now, onupdate=datetime.now)

File: admin/src/api/second-level-task.ts
import request from './request'

export interface SecondLevelTask {
  id: number
  taskId: string
  firstLevelTaskId: string
  stageId: string
  title: string
  description: string
  detail: string
  type: string
  completionType: string
  actionText: string
  actionUrl: string
  /** 行为类型（#PB-39 起投影带出；10 值枚举，未声明 = none） */
  actionType: string
  /** 行为参数（仅 data_form / data_upload 有值；不透明标识） */
  actionParam: string | null
  tag: string
  defaultCompleted: number
  status: number
  sortOrder: number
  createdAt: string
  updatedAt: string
}

export interface CreateSecondLevelTaskParams {
  taskId: string
  firstLevelTaskId: string
  stageId: string
  title: string
  description?: string
  detail?: string
  type: string
  completionType: string
  actionText?: string
  actionUrl?: string
  /** 行为类型（省略 = 落默认 none；非法值后端 400） */
  actionType?: string
  /** 行为参数：data_* 必填非空；其余行为必须为 null/空（显式 null = 清空） */
  actionParam?: string | null
  tag?: string
  defaultCompleted?: number
  sortOrder?: number
}

// 获取二级任务列表
export const getSecondLevelTaskList = (firstLevelTaskId?: string, stageId?: string) => {
  const params: any = {}
  if (firstLevelTaskId) params.firstLevelTaskId = firstLevelTaskId
  if (stageId) params.stageId = stageId
  return request.get<SecondLevelTask[]>('/admin/task/list', { params })
}

// 获取二级任务详情
export const getSecondLevelTaskDetail = (id: number) => {
  return request.get<SecondLevelTask>(`/admin/task/${id}`)
}

// 创建二级任务
export const createSecondLevelTask = (data: CreateSecondLevelTaskParams) => {
  return request.post('/admin/task/create', data)
}

// 更新二级任务
export const updateSecondLevelTask = (id: number, data: Partial<CreateSecondLevelTaskParams>) => {
  return request.put(`/admin/task/${id}`, data)
}

// 删除二级任务
export const deleteSecondLevelTask = (id: number) => {
  return request.delete(`/admin/task/${id}`)
}
